packages/worker/src/api/controllers/global/users.ts

```diff
@@ -23,6 +23,13 @@ const MAX_USERS_UPLOAD_LIMIT = 1000
 
 export const save = async (ctx: any) => {
   try {
+    if (
+      !ctx.request.body._id &&
+      !ctx.internal &&
+      (!ctx.user || !ctx.user.admin || !ctx.user.admin.global)
```

Comment:
I find this block quite hard to read, could it be simplified to:

```ts

const body =  ctx.request.body // re-use this for sdk.users.save
const isCreate = !body._id
const isAdmin = !!ctx.user.admin?.global

// this is a create and we are not admin
if (isCreate && !isAdmin) {
   // throw
}
```

Reply:
if we want to retain the ability for the internal api key to create users no matter what, we could do:

```ts
if ((isCreate && !isAdmin) && !ctx.internal) {
   // throw
}
```

Reply:
should also add a test to ensure the internal api key headers can be used